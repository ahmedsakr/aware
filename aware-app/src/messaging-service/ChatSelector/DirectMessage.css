.direct-message {
    color: white;
    padding: 0;
    margin: 0;
}

.direct-message:hover {
    background-color: #232f3e;
    text-decoration: none;
}

.direct-message-avatar, .direct-message-content {
    display: inline-block;
}

.direct-message-content > p {
  padding: 0;
  margin: 0;
}

.direct-message-content {
    text-align: left;
    display: inline-block;
}
.direct-message-preview {
    border-radius: 5px;
    font-style: italic;
}

@media screen and (max-width: 576px) {
    .direct-message-content > p {
        display: none;
    }

    #vertical-menu > h3 {
        display: none;
    }

    .navbar-item-name {
        display: none;
    }
}

@media screen and (min-width: 576px) and (max-width: 768px) {
    .direct-message-content > p {
        font-size: 0.8em;
    }
}

@media screen and (min-width: 768px) and (max-width: 992px) {
    .direct-message-content > p {
        font-size: 2em;
    }
  }