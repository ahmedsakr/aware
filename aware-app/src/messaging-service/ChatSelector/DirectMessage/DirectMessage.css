.direct-message {
    color: #babdc0;
    padding: 0;
    margin: 0;
}

.direct-message:hover {
    background-color: #ccc;
    text-decoration: none;
}

.direct-message-avatar, .direct-message-content {
    display: inline-block;
}

.direct-message-content > p {
  padding: 0;
  margin: 0;
}

#direct-message-preview {
  padding-left: 10px;
}

.direct-message-content {
    text-align: left;
    display: inline-block;
}
.direct-message-preview {
    border-radius: 5px;
}
