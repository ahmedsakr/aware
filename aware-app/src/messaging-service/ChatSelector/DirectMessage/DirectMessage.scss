.direct-message {
    color: #babdc0;
    padding: 0;
    margin: 0;
    display: flex;
    align-items: center;
}

.direct-message:hover {
    background-color: #232f3e;
    text-decoration: none;
}

.direct-message-content > p {
  padding: 0;
  margin: 0;
}

#direct-message-preview {
  padding-left: 10px;
}

.direct-message-content {
    text-align: left;
    display: inline-block;
}
.direct-message-preview {
    border-radius: 5px;
}
