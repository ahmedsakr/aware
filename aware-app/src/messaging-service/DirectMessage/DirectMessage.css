#direct-message {
    color: white;
    padding: 0;
    margin: 0;
}

.direct-message-avatar, .direct-message-content {
    display: inline-block;
    color: white;
}

.direct-message-content > p {
  padding: 0;
  margin: 0;
}

.direct-message-content > #direct-message-preview {
  padding-left: 10px;
}

.direct-message-content {
    text-align: left;
    display: inline-block;
    vertical-align: middle;
}
.direct-message-preview {
    border-radius: 5px;
    font-style: italic;
}
