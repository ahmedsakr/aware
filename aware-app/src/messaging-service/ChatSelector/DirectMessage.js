import React, { Component } from 'react';
import './DirectMessage.css';

class DirectMessage extends Component {
  render() {
    const { src, name } = this.props;
    return (
      <div class="direct-message">
        <div class="direct-message-avatar col-xs-12 col-sm-2">
          <img src={process.env.PUBLIC_URL + src} alt={name} />
        </div>

        <div class="direct-message-content col-9 pr-0">
          <p>{name}</p>
          <p class="direct-message-preview">Preview of last message...</p>
        </div>
      </div>

    );
  }
}

export default DirectMessage;
