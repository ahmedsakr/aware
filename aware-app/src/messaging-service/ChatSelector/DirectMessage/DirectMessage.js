import React, { Component } from 'react';
import './DirectMessage.scss';

class DirectMessage extends Component {
  render() {
    const { src, name } = this.props;
    return (
      <div class="col-12 direct-message">
        <div class="direct-message-avatar col-2">
          <img src={process.env.PUBLIC_URL + src} alt={name} />
        </div>

        <div class="direct-message-content col-10">
          <p class="col-12">{name}</p>
          <p class="direct-message-preview">Preview of last message...</p>
        </div>
      </div>

    );
  }
}

export default DirectMessage;
