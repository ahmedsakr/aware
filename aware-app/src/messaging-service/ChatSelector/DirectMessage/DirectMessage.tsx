import React from 'react';
import './DirectMessage.css';

type DirectMessageProps = {
    src: string,
    name: string
};

type DirectMessageState = {};

export default class DirectMessage extends React.Component<DirectMessageProps, DirectMessageState> {
    render(): JSX.Element {
        const { src, name } = this.props;
        return (
            <div className="col-12 direct-message">
                <div className="direct-message-avatar col-2">
                    <img src={process.env.PUBLIC_URL + src} alt={name} />
                </div>

                <div className="direct-message-content col-10">
                    <p className="col-12">{name}</p>
                    <p className="direct-message-preview">Preview of last message...</p>
                </div>
            </div>
        );
    }
}
