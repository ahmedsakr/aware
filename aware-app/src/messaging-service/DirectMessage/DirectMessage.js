import React, { Component } from 'react';
import './DirectMessage.css';

class DirectMessage extends Component {
  render() {
    const { src, name } = this.props;
    return (
      <div class="direct-message">
        <div class="direct-message-avatar col-3">
          <img src={process.env.PUBLIC_URL + src} alt={name} />
        </div>

        <div class="direct-message-content col-9">
          <p class="col-12">{name}</p>
          <p class="direct-message-preview">Preview of last message...</p>
        </div>
      </div>

    );
  }
}

export default DirectMessage;
