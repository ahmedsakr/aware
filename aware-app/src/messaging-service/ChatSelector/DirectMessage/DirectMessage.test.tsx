import React from 'react';
import { shallow } from 'enzyme';

import DirectMessage from './DirectMessage';

describe('DirectMessage', () => {
    it('should render correctly', () => {

        const component = shallow(
            <DirectMessage
                src="my-image.png"
                name="John Doe" />
        );

        expect(component).toMatchSnapshot();
    });
});