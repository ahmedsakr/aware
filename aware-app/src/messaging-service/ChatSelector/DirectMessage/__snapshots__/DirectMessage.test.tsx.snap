version https://git-lfs.github.com/spec/v1
oid sha256:f5f86231a4fd53a6327d41f3a4f7a6fd6d584cc41cb31e660aab07e2baf4e28a
size 502
